### DISCUSSION TITLE
Best practice for randomly sampled survey blocks

### DISCUSSION THREAD

**Discussion by RichardWundrack** [rubenarslan/formr.org#542] [Q&A]
Hi formr family,

I was wondering whether there is a recommended best practice for setting up surveys that randomly present a subset of randomly ordered item blocks within whom items may also be randomly ordered? I tried something like in the table here with a variable "random_block" that calculates the blocks to be shown depending on showifs. However, while this seems to work fine as a toy example, set up on a bigger scale – at least in the test runs – it seems to crumble so that items are presented in different blocks as well as "out of the randomized order", i.e. row 1, 2 get mixed up as well ...

<img width="1112" alt="Screenshot 2024-04-22 at 13 48 40" src="https://github.com/rubenarslan/formr.org/assets/38461235/d8aaeb59-d2c5-4904-bd7d-c2d40a599d10">

Thanks for any help or shared experience on the topic!

Best,
Richard

**Comment by rubenarslan** [comment]
I don't think you can refer to "block_order" in the R code, so you'd have to handcode this and enter `"AA" %in jsonlite::fromJSON*(random_block)` etc.
